cartoframes/data/enrichment/points_enrichment.py

```diff
@@ -0,0 +1,95 @@
+import uuid
+
+from . import bigquery_client
+from ...auth import get_default_credentials
+from collections import defaultdict
+
+
+_ENRICHMENT_ID = 'enrichment_id'
+
+# TODO: process column name in metadata, remove spaces and points
+
+
+def enrich_points(data, variables, data_geom_column='geometry', filters=dict()):
+
+    credentials = get_default_credentials()
+
+    bq_client = bigquery_client.BigQueryClient(credentials)
+
+    # Copy dataframe and generate id to join to original data later
+    data_copy = data.copy()
+    data_copy[_ENRICHMENT_ID] = range(data_copy.shape[0])
+    data_geometry_id_copy = data_copy[[_ENRICHMENT_ID, data_geom_column]]
+
+    data_tablename = uuid.uuid4().hex
+
+    bq_client.upload_dataframe(data_geometry_id_copy, _ENRICHMENT_ID, data_geom_column, data_tablename)
+
+    variables_id = variables['id'].tolist()
+    table_to_variables = __process_enrichment_variables(variables_id)
+    table_data_enrichment = list(table_to_variables.keys()).pop()
+    table_geo_enrichment = __get_name_geotable_from_datatable(table_data_enrichment)
+    variables_list = list(table_to_variables.values()).pop()
+
+    filters_str = __process_filters(filters)
+
+    sql = '''
+        SELECT data_table.{enrichment_id},
+              {variables},
+              ST_Area(enrichment_geo_table.geom) AS area,
+              NULL AS population
+        FROM `carto-do-customers.{user_workspace}.{enrichment_table}` enrichment_table
+        JOIN `carto-do-customers.{user_workspace}.{enrichment_geo_table}` enrichment_geo_table
+          ON enrichment_table.geoid = enrichment_geo_table.geoid
+        JOIN `{working_project}.{working_dataset}.{data_table}` data_table
```

Comment:
mmm, are we uploading the user data to a different project/dataset than the user dataset? In authorization time we are granting write permissions over the user dataset specifically to allow uploading data, but not sure what's the best option.

Writing on this `shared` dataset would mean any user would have permissions to write other users tables?

Reply:
Yes, we should talk about this, I think the best way is to create a new project and a dataset for every user who tries to upload something, so we can give permissions to this dataset too. But this dataset needs to be created by the people who are creating the authorize endpoint because if you want to give permissions to a dataset this needs to be created already.

Reply:
> I think the best way is to create a new project and a dataset for every user who tries to upload something

Yep we are already doing that. The service account (or token) has write permissions on `carto-do-customers.{username}` so I think it's safe to upload temp tables there.

See related [issue](https://github.com/CartoDB/cartodb-central/issues/2554#issuecomment-528440295)